torch/csrc/jit/passes/dead_code_elimination.cpp

```diff
@@ -111,7 +120,11 @@ class DeadCodeEliminator {
       // since the mark algorithm should do that. But currently, the marking
       // doesn't reach loop counters in certain cases (see TestScript.test_pass)
       if (!marked_.count(node) && !node->hasUses()) {
-        it.destroyCurrent();
+        if (collect_only_) {
+          dead_nodes_.insert(node);
```

Comment:
Is FindDeadNodes able to find all of the dead nodes? If we have a graph like the following: `a -> b` and b's output isn't used by anything and the only use of `a->output()` is `b`, would FindDeadNodes report `b` or would it report both `a` and `b`?

Reply:
Ohhhh good point... If it was only `marked_.count(...)` it would be ok, but there's this pesky `!node->hasUses()`!

Reply:
Right, and it looks like there is a TODO item to remove the node->hasUses because there is an edge case in marking that doesn't reach it... it might be safe to ignore that for now but a big comment would be good

Reply:
I think I have a different fix.